Source/SwiftyJSON.swift

```diff
@@ -123,6 +123,42 @@ public struct JSON {
         }
         self.init(dictionary)
     }
+    
+    /**
+     Merges another JSON into this JSON, whereas primitive values which are not present in this JSON are getting added, 
+     present values getting overwritten, array values getting appended and nested JSONs getting merged the same way.
+ 
+     - parameter other: The JSON which gets merged into this JSON
+     */
+    public mutating func merge(_ other: JSON) {
+        if self.type == other.type {
+            switch self.type {
+            case .dictionary:
+                for (key, _) in other {
+                    self[key].merge(other[key])
+                }
+            case .array:
+                self = JSON(self.arrayValue + other.arrayValue)
+            default:
+                self = other
+            }
+        } else {
+            self = other
```

Comment:
@danielkiedrowski Sorry for the delay on this.

Oh got it, I didn't realize it would replace nested types too. I actually think nested fields make a lot more sense to be replaced when they have different types.

I would expect this:

```
let a = ["foo": "bar"]
let b = ["foo": 1]
let c = a.merged(with: b) // c.foo == 1
```

but not this:

```
let a = ["foo": "bar"]
let b = [1]
let c = a.merged(with: b) // c == [1]
```

So what I would propose here is to only replace nested fields, but add a sanity type check for the root object. Would that work?